File: java/Median of two sorted array.java
class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        int m = nums1.length;
        int n = nums2.length;
        int[] arr3 = new int[m + n];
        int i = 0, j = 0, k = 0;

        while (i < m && j < n) {
            if (nums1[i] < nums2[j]) {
                arr3[k++] = nums1[i++];
            } else {
                arr3[k++] = nums2[j++];
            }
        }
        while (i < m) {
            arr3[k++] = nums1[i++];
        }
        while (j < n) {
            arr3[k++] = nums2[j++];
        }
        int mid = (m + n) / 2;
        if ((m + n) % 2 == 0) {
            return (arr3[mid - 1] + arr3[mid]) / 2.0;
        } else {
            return arr3[mid];
        }
    }

    public static void main(String[] args) {
        Solution solution = new Solution();

        int[] nums1 = {1, 3};
        int[] nums2 = {2};
        System.out.println(solution.findMedianSortedArrays(nums1, nums2));

        int[] nums1_2 = {1, 2};
        int[] nums2_2 = {3, 4};
      System.out.println(solution.findMedianSortedArrays(nums1_2, nums2_2));
    }
}

File: c++/Array/median_of_two_sorted_array.cpp
#include <iostream>
#include <vector>
#include <algorithm> 

using namespace std;

// Function to find the median of two sorted arrays
double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2) {
    // Step 1: Merge both arrays
    for (int i = 0; i < nums2.size(); i++) {
        nums1.emplace_back(nums2[i]);  // add elements of nums2 to nums1
    }

    // Step 2: Sort the merged array
    sort(nums1.begin(), nums1.end());

    // Step 3: Find the median
    int n = nums1.size();
    int mid = n / 2;

    // If number of elements is even, average of two middle elements
    if (n % 2 == 0) {
        return (nums1[mid] + nums1[mid - 1]) / 2.0;
    }

    // If number of elements is odd, middle element is median
    return nums1[mid];
}

// Main function to test the code
int main() {
    vector<int> nums1 = {1, 3};
    vector<int> nums2 = {2};

    double median = findMedianSortedArrays(nums1, nums2);

    cout << "Median is: " << median << endl;

    return 0;
}
